cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobDetail.java

```diff
@@ -55,11 +64,13 @@ public boolean equals(Object o) {
       return false;
     }
     RuntimeJobDetail that = (RuntimeJobDetail) o;
-    return runInfo.equals(that.runInfo) && status == that.status;
+    return runInfo.equals(that.runInfo)
+        && status == that.status
+        && jobStatusDetails == that.jobStatusDetails;
```

Comment:
You should use `Objects.equals` instead of `==` to compare objects.

Reply:
Done.